tools/mkclass/classcompiler.cpp

```diff
@@ -1482,6 +1482,10 @@ void ClassCompiler::CompileStream(const std::string& path, std::istream& input,
 		<< "#pragma warning( push )" << std::endl
 		<< "#pragma warning( disable : 4244 )" << std::endl
 		<< "#pragma warning( disable : 4800 )" << std::endl
+		<< "#else /* _MSC_VER */" << std::endl
+		<< "#pragma GCC diagnostic push" << std::endl
+		<< "#pragma GCC diagnostic ignored \"-Wunused-parameter\"" << std::endl
+		<< "#pragma GCC diagnostic ignored \"-Wunused-variable\"" << std::endl
```

Comment:
> Disabled warnings for unused parameters/variables on generated files (similar to how it had already been on MSVC, or at least I assume that's what the disabled warnings on MSVC are)

MSVC warnings 4244 and 4800 have nothing to do with unused symbols.

* https://learn.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-2-c4244?view=msvc-170
* https://learn.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-3-c4800?view=msvc-170